src/components/Board.js

```diff
@@ -5,29 +5,115 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+
 
 class Board extends Component {
-  constructor() {
-    super();
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+    boardName: PropTypes.string.isRequired,
+    updateStatusCallback: PropTypes.func.isRequired
+  };
 
+  constructor(props) {
+    super();
     this.state = {
-      cards: [],
+      boardName: props.boardName,
+      cards: []
     };
   }
 
+  getBoardData = (boardName) => {
+    this.props.updateStatusCallback(`Loading cards for ${this.props.boardName}`, 'success');
+    const BOARD_URL = `${this.props.url + boardName}/cards`;
+
+    axios.get(BOARD_URL)
+    .then((response) => {
+      this.props.updateStatusCallback(`Successfully loaded cards for ${boardName}`, 'success');
+      const cardData = response.data.slice(0, 100);
+      this.setState({
+        cards: cardData
+      });
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback(`There was a problem loading cards: ${error.message}`, 'error');
+    })
+  }
+
+  componentDidMount() {
+    this.getBoardData(this.props.boardName);
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    if (nextProps.boardName !== this.state.boardName) {
+      this.setState({
+        boardName: nextProps.boardName
```

Comment:
I feel like there ought to be a way to do this without using the `UNSAFE` function. You might be able to put it in `componentDidUpdate()`, which is like `componentDidMount()` for a re-render after props (or state) changes. It looks like CDU receives the old props and state as arguments - see https://reactjs.org/docs/react-component.html#componentdidupdate

Since which data this component loads and displays is dependent on the props that are passed in, the other option would be to promote this state out of `Board` entirely and have it live in `App`. That breaks your encapsulation though - `App` is now in charge of too many things.